consent/models.py

```diff
@@ -58,7 +62,18 @@ def proxied_get(self, *args, **kwargs):
                 except DataSharingConsent.DoesNotExist:
                     # A record for the course run didn't exist, so modify the query
                     # parameters to look for just a course record on the second pass.
-                    kwargs['course_id'] = parse_course_key(course_run_key)
+
```

Comment:
Since your `.get_course_id()` already accepts a course_id or a course_run_id, we can ditch a lot of the duplicated logic here by just squishing away the first checks and relying on the `get_course_id`. I don't think it's worth a ticket, but maybe just a `NOTE` or `TODO` about cleaning this method up. 

(I don't think the work should be part of this PR, so we can get this over the line)

Reply:
sounds good i'll make a ticket and add a `TODO`